cmd/geth/snapshot.go

```diff
@@ -674,7 +681,14 @@ func traverseRawState(ctx *cli.Context) error {
 	chaindb := utils.MakeChainDatabase(ctx, stack, true, false)
 	defer chaindb.Close()
 
-	triedb := utils.MakeTrieDatabase(ctx, chaindb, false, true)
+	var triedb *trie.Database
```

Comment:
Those `MakeTrieDatabase` codes seems somewhat redundant. Could you define a common function to do this?

Reply:
The MakeTrieDatabase function requires the upper layer to pass in chaindb or separatedb, and is not encapsulated together, which can make the operation of chaindb more independent.